### ISSUE TITLE
Bootstrap 4 Support

### ISSUE THREAD

**Issue by sschueller** [gitana/alpaca#511]
Hi,

What are the plans for bootstrap 4 support? The beta will be out very shortly.

Thanks

**Comment by uzquiano** [comment]
I think we're interested in doing this as our own products use Bootstrap 3 and we'd probably like to move them to 4 at some point.  That said, we don't share a high urgency for this.

It would be interesting to see if someone in the community could take a shot at the templates and layout files needed.  I don't imagine it'd be all that hard.  But there would certainly be a lot of iteration which we could achieve together as a group.

I'm also interested in there are other frameworks we should support.  Our legacy support for jQuery mobile and jQuery UI is fading fast and it's probably time to drop those in favor of newer presentation frameworks.

**Comment by idealley** [comment]
I was thinking about Vuejs

**Comment by biskit** [comment]
ReactJS with a couple of CSS frameworks (SemanticUI/Bootstrap4) would be great choices.

**Comment by idealley** [comment]
@uzquiano it would be great to get rid of jquery altogether and to have a ES6 library that compiling to the browser and nodejs. Thus it could become compatible with server side rendered websites.

If we think that through it could be compatible with `.jsx` and `html` like components (Vue, Angular).

**Comment by petri** [comment]
> it would be great to get rid of jquery altogether 

I agree that in 2019, we can do without jQuery. But Bootstrap 4, which is the topic of this issue, requires it. Although community projects exist to base the JS on something else, I would think that it makes sense for alpacajs to stick to stock bootstrap4 with all its normal dependencies.

Having said that, I would suggest adding dropping of jQuery as its own issue and discuss the topic there. Likewise for any explicit React/Vue/Angular support.

**Comment by calpirro** [comment]
Here Bootstrap 4 suport needed also...

**Comment by JonDavis-AZED** [comment]
It's been 2 1/2 years since the suggestion to switch up to Bootstrap 4. Alpaca is still a key player in JSON Schema based form generator tools, but no one uses Bootstrap 3 in new projects anymore.

Has this effort begun?

**Comment by BlueWater86** [comment]
I've been watching this one for a long time too. We ended up dropping Alpaca from the stack and rolling our own which will start its open source journey after our next product release.

It is powered by the speed of [hyperHTML](https://github.com/WebReflection/hyperHTML) and consumes form definitions that were originally written for Alpaca.

Tested:
![image](https://user-images.githubusercontent.com/6543901/68343027-a0c02f00-0137-11ea-8792-6d4506c1ce6b.png)

Simple class-based field themes:
![image](https://user-images.githubusercontent.com/6543901/68342638-c862c780-0136-11ea-9b76-91fa98c668d4.png)

Explicit element layout:
![image](https://user-images.githubusercontent.com/6543901/68344123-06151f80-013a-11ea-97ac-44e375a24af7.png)

Forms ready for persistence to MongoDB:
![image](https://user-images.githubusercontent.com/6543901/68343522-b84be780-0138-11ea-8a50-9f02a160f262.png)

**Comment by JonDavis-AZED** [comment]
That's nifty, @BlueWater86, and there are lots of other neat libs and architectures out there, but this issue is for migrating from Bootstrap 3 to Bootstrap 4.

My interest in Alpaca was to keep it as a simple client-side library with no back-end dependency like Node. I'm using it with .NET Core in a simple .NET Core 3 app which comes with BS4. I can tolerate jQuery dependency as that is the current design of Alpaca to begin with (and the team I'm on hasn't gotten up to speed with anything else.)

**Comment by JonDavis-AZED** [comment]
I mean, just to clarify, Bootstrap 5 is actively in the works and is actually late to deliver on a beta. 
https://github.com/twbs/release 
https://github.com/twbs/bootstrap

But we're stuck on Bootstrap 3 in Alpaca, nearly two releases behind. I'd jump on this but I'm still new to Alpaca, and for that matter I'm not real deep on knowing Bootstrap

**Comment by BlueWater86** [comment]
Yes, definitely a shameless "nifty" product plug above but I guess that was the point. Looking at the commits landing in Alpaca for 3 months and seeing nothing remotely related to a bootstrap 4 theme along with this issue sitting dormant for 18 months at the point of our decision we bailed on the library. Maybe the team maintaining the library will get a bump from my comment.

P.S. Everything you see above is pure ES6 browser code; the only dependency is the 21KB hyperHTML library.

**Comment by khadgarmage** [comment]
@uzquiano 
Is there any plan to support vue?

**Comment by ndbroadbent** [comment]
@BlueWater86 What's the status of your new open source forms library that supports Bootstrap 4? I'd be very interested in trying it out!

**Comment by BlueWater86** [comment]
@ndbroadbent I haven't moved on publicly hosting & open sourcing the library just yet but do still plan to. I will get the process started tonight.

**Comment by znfgnu** [comment]
@BlueWater86 Any updates about the progress? It would be nice to try your library out.
